### PR TITLE
The culling: Part 2

### PR DESCRIPTION
**Pull request by willkg**
This PR focuses on the fs external stuff. It squashes down the class hierarchy so we end up with `FSRadixTreeStorage` (which becomes the new default) and `FSTemporaryStorage` which is used to queue crashes from the web app for the crashmover app.

It also fixes up some comments and docstrings so the auto-generated docs are better.

r?

### CODE REVIEW

**Comment by willkg** [comment]
The hard part to review here is the part where we squash down the filesystem crashstore classes. We're squashing down a hierarchy that includes a diamond inheritance section. I took all the code from the classes that were going away and put them in the `FSTemporaryStorage` class in reverse MRO order. So we start with a class hierarchy like this (excuse the ascii art):

```
               CrashStoreBase
                     |
             FSRadixTreeStorage
              /              \
FSLegacyRadixTreeStorage     FSDatedRadixTreeStorage
              \               /
        FSLegacyDatedRadixTreeStorage
                     |
              FSTemporaryStorage
```

The reverse MRO for `FSTemporaryStorage` ends up as:

```
[root@60fdafeb7cb4 app]# python
Python 2.7.5 (default, Nov 20 2015, 02:00:19) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-4)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from collector.external.fs.crashstore import FSTemporaryStorage
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named crashstore
>>> from collector.external.fs.crashstorage import FSTemporaryStorage
>>> FSTemporaryStorage.__mro__
(<class 'collector.external.fs.crashstorage.FSTemporaryStorage'>, 
<class 'collector.external.fs.crashstorage.FSLegacyDatedRadixTreeStorage'>, 
<class 'collector.external.fs.crashstorage.FSDatedRadixTreeStorage'>, 
<class 'collector.external.fs.crashstorage.FSLegacyRadixTreeStorage'>, 
<class 'collector.external.fs.crashstorage.FSRadixTreeStorage'>, 
<class 'collector.external.crashstorage_base.CrashStorageBase'>, 
<class 'configman.required_config.RequiredConfig'>, 
<type 'object'>)
>>> list(reversed(FSTemporaryStorage.__mro__))
[<type 'object'>, <class 'configman.required_config.RequiredConfig'>, 
<class 'collector.external.crashstorage_base.CrashStorageBase'>, 
<class 'collector.external.fs.crashstorage.FSRadixTreeStorage'>, 
<class 'collector.external.fs.crashstorage.FSLegacyRadixTreeStorage'>, 
<class 'collector.external.fs.crashstorage.FSDatedRadixTreeStorage'>, 
<class 'collector.external.fs.crashstorage.FSLegacyDatedRadixTreeStorage'>, 
<class 'collector.external.fs.crashstorage.FSTemporaryStorage'>]
>>> 
```

Thus the classes we're squashing in reverse MRO order are these: `FSLegacyRadixTreeStorage`, `FSDatedRadixTreeStorage`, `FSLegacyDatedRadixTreeStorage`, `FSTemporaryStorage`.

I ordered the contents of all the classes in this order in `FSTemporaryStorage`, then removed the top-most attribute/method where there were duplicates. In this way, we end up with a squashed `FSTemporaryStorage` that's exactly like we'd have if we kept the whole hierarchy.

I think everything else in this PR is pretty straight-forward.

@adngdb Can you review this? I think reviewing involves something like:

1. verifying my squashing of `FSTemporaryStorage` is correct
2. checking out the other code changes
3. verifying I didn't remove things I shouldn't have (pretty sure everything I removed shouldn't be in the collector, but I could be wrong)
4. verify Travis is happy

File: docs/crashstorage_fs.rst [hunk 1]
```diff
@@ -7,10 +7,4 @@ Crash storage
 
 .. autoconfigman:: collector.external.fs.crashstorage.FSRadixTreeStorage
 
-.. autoconfigman:: collector.external.fs.crashstorage.FSDatedRadixTreeStorage
```
Comment (adngdb) [inline docs/crashstorage_fs.rst]: Why remove this one?

File: collector/external/fs/crashstorage.py [hunk 2]
```diff
@@ -438,37 +429,37 @@ def new_crashes(self):
                     try:
                         # If the current slot is greater than the hour slot
                         # we're processing, then we can conclude the directory
-                        # is safe to remove.
+                        # is safe to remove
```
Comment (adngdb) [inline collector/external/fs/crashstorage.py]: That is surprising. I thought the convention was to keep periods at the end of sentences in comments. Is it not?

File: docs/crashstorage_fs.rst (hunk 1 above)
Comment (adngdb) [inline docs/crashstorage_fs.rst]: OK, it's been removed in following commits. :)

**Comment by adngdb** [comment]
r+

File: collector/external/fs/crashstorage.py (hunk 2 above)
Comment (willkg) [inline collector/external/fs/crashstorage.py]: I did this a while ago. I think I was making it consistent in this file, but don't remember for sure.

**Comment by willkg** [comment]
Thank you for looking through this! It helps a ton!

**Pull request merged**